ios/engine/KMEI/KeymanEngine/Classes/Migrations.swift

```diff
@@ -393,4 +418,34 @@ enum Migrations {
     }
     return latestVersion
   }
+
+  static func migrateDocumentsFromPreBrowser() throws {
+    log.info("Cleaning Documents folder due to 12.0 installation artifacts")
+
+    // Actually DO it.
+    let documentFolderURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
+    // The String-based version will break, at least in Simulator.
+    let contents = try FileManager.default.contentsOfDirectory(at: documentFolderURL, includingPropertiesForKeys: nil, options: [])
+
+    try contents.forEach { fileURL in
+      // 12.0: extracts .kmp files by first putting them in Documents and giving
+      // them this suffix.
+      if fileURL.lastPathComponent.hasSuffix(".kmp.zip") {
+        // Renames the .kmp.zip files back to their original .kmp filename.
+        let destFile = fileURL.lastPathComponent.replacingOccurrences(of: ".kmp.zip", with: ".kmp")
+        let destURL = fileURL.deletingLastPathComponent().appendingPathComponent(destFile)
+
+        log.debug("\(fileURL) -> \(destURL)")
+        try FileManager.default.moveItem(at: fileURL, to: destURL)
```

Comment:
What happens if there is a `foo.kmp.zip` and a `foo.kmp` in the folder?

Reply:
Because of how KMP installation worked in previous versions of the app, this state should be impossible.  We never stored raw .kmps in that folder.

If, somehow, one _does_ exist, the attempt to move the item will likely fall over with an error and abort the file-relocation process, which would then result in an error message, but not an app crash.  (See line 95 for the `catch` block that handles this scenario.)